### PR TITLE
New live submission is not appended and verdict is not updated

### PR DESCRIPTION
**Pull request by Wally5077**

### CODE REVIEW

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Code works well.

Just one change request:
Query all problems/students at once in the query.

File: src/services/LiveSubmissionsService.js [hunk 1]
```diff
@@ -18,20 +18,38 @@ export class LiveSubmissionsService extends AbstractService {
...
+                .then(events => subscriber(events[0])));
     }
 
-    async completeLiveSubmissionFields(liveSubmissionEvent) {
-        liveSubmissionEvent.problemTitle = (await this.problemService.getProblemById(liveSubmissionEvent.problemId)).title;
-        liveSubmissionEvent.studentName = (await this.studentService.getStudentById(liveSubmissionEvent.studentId)).name;
-        return Promise.resolve(liveSubmissionEvent);
+    async completeLiveSubmissionsFields(liveSubmissions) {
+        const problemIds = liveSubmissions.map(liveSubmission => liveSubmission.problemId)
+            .filter((value, index, self) => self.indexOf(value) === index);
```
Comment (Johnny850807) [inline src/services/LiveSubmissionsService.js:41]: extract a  `distinct(array)` method

File: src/services/LiveSubmissionsService.js [hunk 2]
```diff
@@ -18,20 +18,38 @@ export class LiveSubmissionsService extends AbstractService {
...
+        const problems = await this.problemService.getProblemsByIds(problemIds);
+
+        const studentIds = liveSubmissions.map(liveSubmission => liveSubmission.studentId)
+            .filter((value, index, self) => self.indexOf(value) === index);
```
Comment (Johnny850807) [inline src/services/LiveSubmissionsService.js:45]: ditto

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

**Review by nyngwang — APPROVED** [review]
LGTM.

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**